# AS event analysis

## differential splicing analysis

### dsflow

**dsflow** is wrapper of AS events inferring, PSI calculation,  differential splicing analysis and significants differential result selection.

Arguments：

* -od: output dirctory
* -md: metadata json, the output of **meta**
* -gtf: genome annotation GTF file
* -et: AS event type [ALL|SE|A5|A3|MX|RI|AF|AL]
* -m: the method used to differential splicing computation [empirical|classical]
* -p：p-value threshold
* -adpsi: absolute dPSI threshold

code：

```bash
$ mkdir result
$ astk dsflow -od result/fb_e11_based -md metadata/fb_e11_based.json \
    -gtf gencode.vM25.annotation.gtf -et ALL

$ ls result/fb_e11_based
dpsi  psi  ref  sig01  tpm
```

output contains 4 directories:

* ref is used to save AS events references files
* tpm is used to save transcript TPM files
* psi is used to save AS event PSI files
* dpsi is used to save AS event dPSI files
* sig01 is used to save significants differential result

### single-step running

We implement the functions of AS events inferring, PSI calculation,  differential splicing analysis based on SUPPA2 algorithm within **ASTK**. Therefore, there are three sub-commands with the same name as SUPPA2. Most of the time, it is more convenient to use [dsflow](en/content/AS_event?id=dsflow).

#### AS events inferring

**generateEvent** is used to infer potential AS events based on genome annotation.

Arguments:

* -gtf: genome annotation GTF file
* -et: AS event type  [ALL|SE|A5|A3|MX|RI|AF|AL]
* --split: split AS events according to whether they overlap with the first or last exon of the transcript [inner|FTE|LTE]
* -o: output path

code：

```bash
$ astk generateEvent -gtf gencode.vM25.annotation.gtf  -et SE \
    -o gencode.vM25

$ head gencode.vM25_SE_strict.ioe
seqname gene_id event_id        alternative_transcripts total_transcripts
chr1    ENSMUSG00000025900.13   ENSMUSG00000025900.13;SE:chr1:4293012-4311270:4311433-4351910:- ENSMUST00000208660.1    ENSMUST00000208660.1,ENSMUST00000208793.1
chr1    ENSMUSG00000025902.13   ENSMUSG00000025902.13;SE:chr1:4492668-4493100:4493466-4493772:- ENSMUST00000027035.9    ENSMUST00000027035.9,ENSMUST00000192650.5,ENSMUST00000191647.1
chr1    ENSMUSG00000025902.13   ENSMUSG00000025902.13;SE:chr1:4492668-4493100:4493490-4493772:- ENSMUST00000116652.7    ENSMUST00000116652.7,ENSMUST00000192650.5,ENSMUST00000191647.1
```

#### AS events PSI calculation

**psiPerEvent** is used to compute AS event PSI values based on transcript TPM.

Arguments:

* -o: output path
* -qf: transcript quantification files
* -ioe: AS events ioe file. it is generated by **generateEvent**
* --tpmThreshold: transcript TPM threshold

code:

```bash
$ astk psiPerEvent -o fb_SE_e11.psi \
    -qf data/quant/fb_e11.5_rep*/quant.sf \
    -ioe gencode.vM25_SE_strict.ioe

$ astk psiPerEvent -o fb_SE_e16.psi \
    -qf data/quant/fb_e16.5_rep*/quant.sf \
    -ioe gencode.vM25_SE_strict.ioe


$ head fb_SE_e11.psi -n 5
event_id        fb_e11.5_rep1   fb_e11.5_rep2
ENSMUSG00000025900.13;SE:chr1:4293012-4311270:4311433-4351910:- 1.0     0
ENSMUSG00000025902.13;SE:chr1:4492668-4493100:4493466-4493772:- 0.5909638662971756      0.7763651621731414
ENSMUSG00000025902.13;SE:chr1:4492668-4493100:4493490-4493772:- 0.8259026936895806      0.8056399646246479
ENSMUSG00000025902.13;SE:chr1:4493863-4495136:4495942-4496291:- 0.21779305143835212     0.3238071557979279


# it also produce the transcript TPM file
$ head fb_SE_e11.tpm -n 3
fb_e11.5_rep1   fb_e11.5_rep2
ENSMUST00000193812.1    0.060544        0.0
ENSMUST00000082908.1    0.0     0.0

```

#### AS events differential splicing computation

**diffSplice** is used to calculate differential splicing for AS events.

Arguments:

* -psi: psi files, the PSI values of replicate samples need to be in one file
* -exp: transcript tpm files, the transcript TPM of replicate samples need to be in one file
* -ref: AS events references ioe file
* -o: output path
* -m: differential splicing method [empirical|classical]，default=empirical

code:

```bash
$ astk diffSplice -psi fb_SE_e11.psi fb_SE_e16.psi\
    -exp fb_SE_e11.tpm fb_SE_e16.tpm\
    -ref gencode.vM25_SE_strict.ioe \
    -o fb_SE_e10_e16.dpsi 
```

## AS events process

### dPSI filtering

**sigFilter** filters differential splicing AS events according to dPSI and p-value. It supports output of rMATS and SUPPA22. **sf**  is the short alias of **sigFilter**.

Arguments:

* -i: dpsi file
* -o: output path
* -adpsi: absolute dPSI threshold
* -p: p-value, default is 0.05
* -dpsi: dPSI threshold, default is 0
* -sep: if set, output will be splited two file according to dPSI > 0 and dPSI < 0

### PSI filtering

**psiFilter** filters AS events with PSI value. **pf** is the short alias of **psiFilter**.

Arguments:

* -i: psi file
* -o: output path
* -minv: minimum PSI threshold value
* -maxv: maximal PSI threshold value
* -minq: minimum quantile threshold value
* -maxq: maximal quantile threshold value

code:

```bash
$ astk pf -i result/fb_e11_based/psi/fb_e11_p0_SE_case.psi \
    -minv 0.8 -o result/fb_e11_based/psi/fb_p0_SE_high.psi
$ astk pf -i result/fb_e11_based/psi/fb_e11_p0_SE_case.psi \
    -maxv -0.2 -o result/fb_e11_based/psi/fb_p0_SE_low.psi
```

### length cluster

**lenCluster** clusters AS events based on alternative exon length. **lc** is the short alias of **lenCluster**.

Arguments:

* -i: input file
* -lr: length range
* -od: output directory

code:

```bash
$ astk lc -i result/fb_e11_based/sig01/*dpsi -lr 1 51 251 1001 \
    -od result/fb_e11_based/lenc

$ ls result/fb_e11_based/lenc
1-51  251-1001  51-251
```

## AS events analysis

### AS events distribution

#### barplot

**barplot** could use to show AS events number distribution of different condition or AS type.

Arguments:

* -i: input files
* -o: output figure path
* -dg: if set, AS events will be classified into two groups(group +: dPSI > 0, group -: dPSI < 0)
* -xl: x labels

code:

```bash
astk barplot -i result/fb_e11_based/sig01/fb_e11_p0_*.sig.dpsi \
    -o img/fb_e11_p0_bar.png -dg -xl A3 A5 AF AL MX RI SE

```

<img src='static/img/fb_e11_p0_bar.png' alt="fb_e11_p0_bar.png"></img>

#### upset plot

**upset** is used to show the intersection of different groups of AS events.

Arguments:

* -i: input files
* -o: output figure path
* -xl: x labels
* -dg: if set, AS events will be classified into two groups(group +: dPSI > 0, group -: dPSI < 0)

code:

```bash
$ astk upset -i result/fb_e11_based/sig01/fb_e11_12_SE.sig.dpsi \
    result/fb_e11_based/sig01/fb_e11_14_SE.sig.dpsi \
    result/fb_e11_based/sig01/fb_e11_16_SE.sig.dpsi \
    -o img/fb_upset.png -xl e11_12 e11_14 e11_16 -dg

```

<img src='static/img/fb_upset_dg.png' alt="fb_upset_dg.png"></img>

### PSI/dPSI pattern analysis

#### PCA

**pca** draws PCA plot using AS events PSI.

Arguments:

* -i: psi files
* -o: output figure path
* --width: figure width
* --height: figure height
* -gl: group labels
* -gb: this option is used to choose whether the sample information is stored in rows or columns;AS event PSI use col, feature values use row

code:

```bash
$ astk pca -i result/fb_e11_based/psi/fb_e11_12_SE_c1.psi \
    result/fb_e11_based/psi/fb_e11_1[2-6]_SE_c2.psi \
    result/fb_e11_based/psi/fb_e11_p0_SE_c2.psi \
    -o img/fb_pca.png --width 6 --height 4 \
    -gl fb_e11.5 fb_e12.5 fb_e13.5 fb_e14.5 fb_e15.5 fb_e16.5 fb_p0 -gb col

```

<img src='static/img/fb_pca.png' alt="fb_pca.png"></img>

#### heatmap plot

**heatmap** draws PSI heatmap plot using PSI values.. **hm** is the short alias of **heatmap**.

Arguments:

* -i: psi files
* -o: output figure path
* -ff: figure format

code:

```bash
$ astk hm -i result/fb_e11_based/sig01/psi/fb_e11_12_SE_c1.sig.psi \
    result/fb_e11_based/sig01/psi/fb_e11_1*_SE_c2.sig.psi \
    -o img/fb_hm.png -ff png

```

<img src='static/img/fb_hm.png' alt="fb_hm.png"></img>

#### volcano plot

**volcano** draws volcano plot using dPSI values.  **vol** is the short alias of **volcano**.

Arguments:

* -i: dpsi file
* -o: output figure path

code:

```bash
$ astk volcano -i result/fb_e11_based/dpsi/fb_e11_p0_SE.dpsi \
    -o img/fb_e11_p0_SE.vol.png 

```

<img src='static/img/fb_e11_p0_SE.vol.png' alt="fb_e11_p0_SE.vol.png"></img>
