# Sequence feature

## Sequence feature extraction

### Splice site strength

**ASTK** re-implemented splice site strength computation based on the [MaxEntScan](http://hollywood.mit.edu/burgelab/maxent/download/fordownload/) (Yeo and Burge,2004) Perl scripts.

**spliceScore** is the sub-command for compute the splice site strength. **sss** is the short alias.

Arguments:

* -e: AS event file
* -od: output directory
* -fi: genome fasta
* -p: process number, default is 4

code

```bash
$ mkdir sss
$ astk sss -e result/fb_e11_based/psi/fb_p0_SE_high.psi -od sss/fb_p0_SE_high -fi GRCm38.primary_assembly.genome.fa

$ tree sss/fb_p0_SE_high
├── A0_5SS
│   ├── A0_5SS.bed
│   └── A0_5SS.fa
├── A1_3SS
│   ├── A1_3SS.bed
│   └── A1_3SS.fa
├── A2_5SS
│   ├── A2_5SS.bed
│   └── A2_5SS.fa
├── A3_3SS
│   ├── A3_3SS.bed
│   └── A3_3SS.fa
├── splice_scores_box.png
└── splice_scores.csv

4 directories, 10 files
```

In the output, the *.bed files store coordinates of splice site regions(for 3' splice-site, 20 nt from the 3' end of the upstream introns and the 3 nt of the exonic sequences; for 5' splice-site, the first 6 nt at the 5' end of the downstream intron, and the 3 nt of the exonic sequences).
The *.fa files store the fasta sequence of the corresponding region, and the sequence of the reversed strand are the reverse complementary sequence.

splice_scores.csv is CSV file to store splice site strength
splice_scores_box.png is boxplot to show splice site strength

<img src='static/img/splice_scores_box.png' alt="splice_scores_box.png"></img>

### GC content

**gcc** is used to compute GC content of each splice-site flank sequence.

Arguments:

* -e: AS event path
* -od: output directory
* -fi: genome fasta
* -bs: bin size or slide window size
* -ef: exon flank width
* -if: intron flank width

code:

```bash
$ mkdir gcc
$ astk gcc -e result/fb_e11_based/psi/fb_p0_SE_high.psi \
    -od gcc/fb_p0_AF_high -ef 150 -if 150 -bs 75 \
    -fi GRCm38.primary_assembly.genome.fa

$ tree gcc/fb_p0_AF_high
├── A1_5SS
│   ├── A1_5SS_dws.bed
│   ├── A1_5SS_dws.fa
│   ├── A1_5SS_dws_gcc.csv
│   ├── A1_5SS_ups.bed
│   ├── A1_5SS_ups.fa
│   └── A1_5SS_ups_gcc.csv
├── A2_3SS
│   ├── A2_3SS_dws.bed
│   ├── A2_3SS_dws.fa
│   ├── A2_3SS_dws_gcc.csv
│   ├── A2_3SS_ups.bed
│   ├── A2_3SS_ups.fa
│   └── A2_3SS_ups_gcc.csv
├── A3_5SS
│   ├── A3_5SS_dws.bed
│   ├── A3_5SS_dws.fa
│   ├── A3_5SS_dws_gcc.csv
│   ├── A3_5SS_ups.bed
│   ├── A3_5SS_ups.fa
│   └── A3_5SS_ups_gcc.csv
├── A4_3SS
│   ├── A4_3SS_dws.bed
│   ├── A4_3SS_dws.fa
│   ├── A4_3SS_dws_gcc.csv
│   ├── A4_3SS_ups.bed
│   ├── A4_3SS_ups.fa
│   └── A4_3SS_ups_gcc.csv
└── gcc.png

4 directories, 25 files  
```

<img src='static/img/gcc.png' alt="gcc.png"></img>

### exon/intron length

**elen** is used to compute elment length of adjacent splice sites.

Arguments:

* -e: AS event path
* -od: output directory
* -log: log2 transformation

code:

```basdh
$ mkdir len
$ astk elen -e result/fb_e11_based/psi/fb_e11_p0_AF_high.psi -od len/fb_p0_AF -log
$ ls len/fb_p0_A
element_len.csv  element_len.png
```

<img src='static/img/element_len.png' alt="element_len.png"></img>

## Sequence feature comparison

### Splice site strength comparison

Splice site strength comparison among multiple condition. it can achieve by **vcmp** sub-command.

Arguments:

* -e: AS event splice-site strength csv files, generated by **sss**
* -o: output path
* -test: statistical test methods, default is Mann-Whitney [Mann-Whitney|t-test_ind|t-test_welch|Wilcoxon]
* -gn: group names
* -ft: figure type, default is box [point|strip|box|boxen|violin|bar]
* -ff: figure outputformat [auto|png|pdf|tiff|jpeg]
* --xtitle: x title
* --ytitle: y title

code:

```bash
$ mkdir vcmp
$ astk vcmp -e sss/fb_e16_SE_*/splice_scores.csv \
    -gn high low -o vcmp/fb_e16_AF_sss.png  \
    --xtitle "splice_site" --ytitle "score" >  vcmp/fb_e16_AF_sss.txt
$ cat vcmp/fb_e16_AF_sss.txt   
p-value annotation legend:
      ns: p <= 1.00e+00
       *: 1.00e-02 < p <= 5.00e-02
      **: 1.00e-03 < p <= 1.00e-02
     ***: 1.00e-04 < p <= 1.00e-03
    ****: p <= 1.00e-04

A1_5SS_high vs. A1_5SS_low: Mann-Whitney-Wilcoxon test two-sided, P_val:1.243e-17 U_stat=6.122e+06
A0_pse_3SS_high vs. A0_pse_3SS_low: Mann-Whitney-Wilcoxon test two-sided, P_val:9.558e-01 U_stat=5.453e+06
A2_pse_3SS_high vs. A2_pse_3SS_low: Mann-Whitney-Wilcoxon test two-sided, P_val:3.981e-01 U_stat=5.515e+06
A3_5SS_high vs. A3_5SS_low: Mann-Whitney-Wilcoxon test two-sided, P_val:2.394e-26 U_stat=4.611e+06
A4_3SS_high vs. A4_3SS_low: Mann-Whitney-Wilcoxon test two-sided, P_val:4.759e-01 U_stat=5.392e+06
```

<img src='static/img/fb_e16_AF_sss.png' alt="fb_e16_AF_sss.png"></img>

### exon/intron length comparison

exon/intron length comparison among multiple condition. it can achieve by **vcmp** sub-command.

* -e: AS event element length csv files, generated by **elen**
* -o: output path
* -test: statistical test methods, default is Mann-Whitney [Mann-Whitney|t-test_ind|t-test_welch|Wilcoxon]
* -gn: group names
* -ft: figure type, default is box [point|strip|box|boxen|violin|bar]
* -ff: figure outputformat [auto|png|pdf|tiff|jpeg]
* --xtitle: x title
* --ytitle: y title
* -facet: facet show
* -xlabel: x labels
* -log: log2 transformation

code:

```bash
astk vcmp -e Length/0hr_FT_se_*/element_len.csv \
        -gn high low -o vcmp/FT_se_lencmp.png -ft box \
        -log --xlabel ups_intron exon dws_intron \
        --xtitle "element" --ytitle "log2(length)" >  vcmp/FT_se_len.txt
```

<img src='static/img/FT_se_lencmp.png' alt="FT_se_lencmp.png"></img>

### GC content comparison

Splice site flank sequence GC content comparison among multiple condition. it can achieve by **vcmp** sub-command.

Arguments:

* -e: AS event splice-site flank sequence GC content csv files, generated by **gcc**, not support slide window
* -o: output path
* -test: statistical test methods, default is Mann-Whitney [Mann-Whitney|t-test_ind|t-test_welch|Wilcoxon]
* -gn: group names
* -ft: figure type, default is box [point|strip|box|boxen|violin|bar]
* -ff: figure outputformat [auto|png|pdf|tiff|jpeg]
* --xtitle: x title
* --ytitle: y title
* -facet: facet show
* -xlabel: x labels
* -log: log2 transformation

code:

```bash
astk vcmp -e gcc/fb_p0_SE_*_b0/gcc.csv --facet \
        -gn high low -o vcmp/gcc/p0_se_gcccmp.png -ft box \
         --xtitle "splic_site" --ytitle "GCC" 
```

<img src='static/img/p0_se_gcccmp.png' alt="p0_se_gcccmp.png"></img>

<h2>References</h2>
<p>
Yeo,G. and Burge,C.B. (2004) Maximum Entropy Modeling of Short Sequence Motifs with Applications to RNA Splicing Signals. 18.
</p>
